plugins/swf-backend/src/service/router.ts

```diff
@@ -187,31 +198,39 @@ function setupInternalRoutes(
     res.status(200).json(processInstance);
   });
 
+  router.delete('/workflows/:swfId', async (req, res) => {
+    const swfId = req.params.swfId;
+    await workflowService.deleteWorkflowDefinitionById(swfId);
+    res.status(201).send();
+  });
+
   router.post('/workflows', async (req, res) => {
     const url = req.query.url;
     const swfData = req.body;
-    let createdWorkflow;
-    if (url && typeof url === 'string' && url.includes(`http`)) {
-      createdWorkflow = await workflowService.saveWorkflowDefinitionFromUrl(
-        url,
-      );
+
+    if (url && url.includes(`http`)) {
+      await workflowService.saveWorkflowDefinitionFromUrl(url);
     } else {
-      createdWorkflow = await workflowService.saveWorkflowDefinition(swfData);
+      await workflowService.saveWorkflowDefinition(swfData);
     }
 
     const swfItem: SwfItem = {
-      id: createdWorkflow.id,
-      definition: JSON.stringify(createdWorkflow),
+      id: swfData.id,
+      definition: JSON.stringify(swfData),
       name: ``,
       description: ``,
     };
     res.status(201).json(swfItem).send();
   });
 
-  router.delete('/workflows/:swfId', async (req, res) => {
-    const swfId = req.params.swfId;
-    await workflowService.deleteWorkflowDefinitionById(swfId);
-    res.status(201).send();
+  router.get('/actions/schema', async (_, res) => {
+    const openApi = await openApiService.generateOpenApi();
+    res.json(openApi).status(200).send();
+  });
+
+  router.put('/actions/schema', async (_, res) => {
```

Comment:
What is the purpose of implementing `put`?

The `get` generates it dynamically on the fly; so saving it somewhere is of no use?

Reply:
get generates the openapi dynamically, put is just to generate + persist the file in the /resources folder mapped in the devmode container, anyway just in case we need to update the openapi file, I used it just in the first time to avoid a manual step for this, anyway we can remove it